src/index/CompressedRelation.cpp

```diff
@@ -623,10 +631,18 @@ DecompressedBlock CompressedRelationReader::readPossiblyIncompleteBlock(
   AD_CORRECTNESS_CHECK(ADDITIONAL_COLUMN_GRAPH_ID <
                        blockMetadata.offsetsAndCompressedSize_.size());
 
-  // We first scan the complete block including ALL columns.
+  bool manuallyDeleteGraphColumn = scanConfig.graphFilter_.deleteGraphColumn_;
+  // We first scan the complete block including ALL columns with the following
```

Comment:
I think I now understand this comment, but the information on why we need ALL columns, and not just the ones that match the original scan specification is missing.

Reply:
That is a real big mess that cannot be understood here individually. I will postpone refactoring this to a separate PR.